homework_3/toDoCLI.js

```diff
@@ -0,0 +1,206 @@
+console.log(home);
+rl.prompt();
+
+// Todo Class
+class ToDo {
+  constructor(id, name, completed) {
+    (this.id = id), (this.name = name), (this.completed = completed);
+  }
+}
+
+class cliFunctions {
+  // Initialize a constructor for ToDo List
+  constructor() {
+    // this.todos = fileName ? ['seedData'] : [];
+    // this.todos = fileName ? 'yes' : 'no';
+    //this.todos = [];
+    // console.log(fileName);
+    this.executedOnce = false;
+    this.todos = [];
+    this.count = 0;
+  }
+  viewTodos() {
+    const todoArr = this.todos;
+    todoArr.forEach((todo) => {
+      const { id, name, completed } = todo;
+
+      console.log(`${id} ${completed ? `[\u2713]` : `[ ]`} ${name}`);
+    });
+    console.log(menuMessage);
+    return this.todos;
+  }
+  addTodo() {
+    rl.prompt();
+    rl.question(`What? \n`, (result) => {
+      const newTodo = new ToDo(this.count, result.trim(), false);
+      this.todos.push(newTodo);
+      console.log(`Created new todo successfully`);
+      console.log(menuMessage);
+      this.count += 1;
+      return;
+    });
+  }
+  completeTodo(input) {
+    const index = parseInt(input.slice(1, input.length));
+
+    const idMap = this.todos.map((todo) => {
+      return todo.id;
+    });
+
+    if (input.slice(0, 1) === 'c' && idMap.includes(index)) {
+      this.todos[index].completed = true;
+      console.log(`Completed "${this.todos[index].name}"`);
+    }
+    console.log(menuMessage);
+    return;
+  }
+  deleteTodos(input) {
+    const idMap = this.todos.map((todo) => {
+      return todo.id;
+    });
+
+    const index = parseInt(input.slice(1, input.length));
+
+    if (input.slice(0, 1) === 'd' && idMap.includes(index)) {
+      console.log(`Deleted "${this.todos[index].name}"`);
+      this.todos.splice(index, 1);
+    } else {
+      console.log(
+        `You entered ${input} and the id ${input.slice(
+          1
+        )} does not exist in the database`
+      );
+    }
+    console.log(menuMessage);
+  }
+  save() {
+    rl.question(`Where? (myTodos.json)`, (answer) => {
+      const path = answer || `${__dirname}/myTodos.json`;
+
+      fs.writeFile(path, JSON.stringify(this.todos), (err) => {});
+
+      console.log(`List saved to "myTodos.json"`);
+      console.log(menuMessage);
+    });
+  }
+  quit() {
+    console.log(`Quitting the Todo CLI program... \nSee you soon! 😄`);
+    rl.close();
+  }
+  cliMenu() {
+    console.log(menuMessage);
+    rl.on('line', (input) => {
+      if (input === 'v') {
+        newCli.viewTodos(input);
+      } else if (input === 'n') {
+        newCli.addTodo(input);
+      } else if (input.slice(0, 1).includes('c')) {
+        newCli.completeTodo(input);
+      } else if (input.slice(0, 1).includes('d')) {
+        newCli.deleteTodos(input);
+      } else if (input === 's') {
+        newCli.save();
+      } else if (input === 'q') {
+        newCli.quit(input);
+      }
```

Comment:
would be a nice idea to have an `else` block to handle commands that are not supported and return a message to the user.